### ISSUE TITLE
Stream closed in another thread (IOError)

### ISSUE THREAD

**Issue by kublaios** [ianrhamilton/fastlane-plugin-saucectl#9]
When running `sauce_runner`, the command fails with the following error:

```
Running version 0.150.0
20:27:55 INF Running XCUITest in Sauce Labs
20:27:55 INF Archiving project.app to .ipa
20:28:00 INF Archiving project-e2e-tests-Runner.app to .ipa
20:28:01 INF Checking if /var/folders/bl/wbxjgtzx7j5_mjsmfr3ynlc00000gp/T/project-1311811755.ipa has already been uploaded previously
20:28:01 INF Checksum: 0b1c55dcc101dcd58e9bd4460584222b592ec79d3b370d28ff36742334f372fd
+---------------------------+---------------------------------------------------------------+
|                                       Lane Context                                        |
+---------------------------+---------------------------------------------------------------+
| DEFAULT_PLATFORM          | ios                                                           |
| PLATFORM_NAME             |                                                               |
| LANE_NAME                 | upload_to_sauce                                               |
| KEYCHAIN_PATH             | ~/Library/Keychains/fastlane_tmp_keychain                     |
| ORIGINAL_DEFAULT_KEYCHAIN | "/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db" |
+---------------------------+---------------------------------------------------------------+
#<Thread:0x00007fc871afe240 /Users/distiller/project/.bundle/ruby/3.0.0/gems/fastlane-plugin-saucectl-0.1.4/lib/fastlane/plugin/saucectl/helper/runner.rb:27 run> terminated with exception (report_on_exception is true):
/Users/distiller/project/.bundle/ruby/3.0.0/gems/fastlane-plugin-saucectl-0.1.4/lib/fastlane/plugin/saucectl/helper/runner.rb:28:in `each': stream closed in another thread (IOError)
	from /Users/distiller/project/.bundle/ruby/3.0.0/gems/fastlane-plugin-saucectl-0.1.4/lib/fastlane/plugin/saucectl/helper/runner.rb:28:in `block (2 levels) in system'
```

This only happens on CI. Following the same steps locally works fine. Coincidentally, we started getting this failure around the time SauceLabs had [downtime](https://twitter.com/sauceops/status/1673329155960713219), but support claims that they resolved all the issues. What's the best way to fix or debug this issue?

**Comment by ianrhamilton** [comment]
@kublaios apologies I just saw this. Are you still having this issue? Seems fine from my side.

**Comment by kublaios** [comment]
> @kublaios apologies I just saw this. Are you still having this issue? Seems fine from my side.

No worries. We could not find the root cause. Although we did not change anything in our setup, our test runs were either timing out or not showing up. This happened after SauceLabs had an outage, but they declined that the issue was on their end. Eventually, we moved to another platform.

Thanks for your work here. I'll go ahead and close the issue.
